app/queries/shortlist_foster_parent_query.rb

```diff
@@ -0,0 +1,43 @@
+class ShortlistFosterParentQuery
+  def initialize(placement_types: [], latitude: nil, longitude: nil, search_radius: nil)
+    @placement_types = placement_types
+    @latitude = latitude
+    @longitude = longitude
+    @search_radius = search_radius
+  end
+
+  def call
+    FosterParent
+      .joins(:placement_suitability)
+      .includes(:placement_suitability)
+      .yield_self(&method(:placements_part))
```

Comment:
could this also be written as `yield_self(&placements_part)` ?

Reply:
No, the method expects a block and this is the only way to "convert" methods into blocks. To allow for the syntax you're writing, we'd need to define something like `placements_patrt = Proc.new {}` but that would loose the point of readability within the class. 

I tried to see if that can be achieved, but this is the only way I found working 🤔

Reply:
Yes , that makes sense